02-oop-inheritance/Observer pattern/script.js

```diff
@@ -0,0 +1,66 @@
+var Movie;
+var MovieObserver;
+
+//metodos para publicar y subscribir
+var EventBus = {
+  topics: {},
+
+  subscribe: function(topic, listener) {
+    // create the topic if not yet created
+    if(!this.topics[topic]) this.topics[topic] = [];
+
+    // add the listener
+    this.topics[topic].push(listener);
+  },
+
+  publish: function(topic, data) {
+    // return if the topic doesn't exist, or there are no listeners
+    if(!this.topics[topic] || this.topics[topic].length < 1) return;
+
+    // send the event to all listeners
+    this.topics[topic].forEach(function(listener) {
+      listener(data || {});
+    });
+  }
+};
+
+
+//Mis dos clases con sus atributos y funciones
+Movie = function (title, year) {
```

Comment:
A bit confusing to define ``Movie`` at the top and then assigning it here. Maybe you could do ``var Movie`` directly in this line. :)